TweetMate/src/main/java/com/ybsystem/tweetmate/activities/VideoActivity.java

```diff
@@ -0,0 +1,129 @@
+
+import static com.ybsystem.tweetmate.resources.ResString.*;
+
+public class VideoActivity extends ActivityBase {
+
+    // Video player
+    private ExoPlayer mExoPlayer;
+    private PlayerView mPlayerView;
+    private PlayerControlView mPlayerControlView;
+
+    // Media
+    private TwitterMediaEntity mMediaEntity;
+    private MediaSource mMediaSource;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Init
+        mMediaEntity = (TwitterMediaEntity)
+                getIntent().getSerializableExtra("MEDIA_ENTITY");
+
+        // Create media
+        Uri videoURI = Uri.parse(PrefSystem.getVideoByQuality(mMediaEntity));
+        mMediaSource = new ProgressiveMediaSource
+                .Factory(new DefaultHttpDataSource.Factory())
+                .createMediaSource(MediaItem.fromUri(videoURI));
+
+        // Set actionbar title
+        switch (mMediaEntity.getType()) {
+            case "video":
+                getSupportActionBar().setTitle(STR_MOVIE);
+                break;
+            case "animated_gif":
+                getSupportActionBar().setTitle(STR_GIF);
+                break;
+        }
+
+        // Set view
+        setContentView(R.layout.activity_video);
+        setPlayerView();
+        setPlayerListener();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (mExoPlayer != null) {
+            mExoPlayer.setPlayWhenReady(false);
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        if (mExoPlayer != null) {
+            mExoPlayer.stop();
+            mExoPlayer.release();
+        }
+    }
+
+    private void setPlayerView() {
+        // Create player
+        mExoPlayer = new SimpleExoPlayer.Builder(this).build();
+        mExoPlayer.setMediaSource(mMediaSource);
+        mExoPlayer.prepare();
+
+        // Set to playerView
+        mPlayerView = findViewById(R.id.player_view);
+        mPlayerView.setPlayer(mExoPlayer);
+
+        // Set to controlView
+        mPlayerControlView = findViewById(R.id.player_control_view);
+        mPlayerControlView.setPlayer(mExoPlayer);
+
+        // Start video and Hide control
+        mExoPlayer.setPlayWhenReady(true);
+        mPlayerControlView.hide();
+    }
+
+    private void setPlayerListener() {
+        // When click video, toggle control visibility
+        mPlayerView.getVideoSurfaceView().setOnClickListener(view -> {
+            if (mPlayerControlView.isVisible()) {
+                mPlayerControlView.hide();
+            } else {
+                mPlayerControlView.show();
+            }
+        });
+
+        // When completed video
+        mExoPlayer.addListener(new ExoPlayer.Listener() {
+            @Override
+            public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+                switch(playbackState) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs32/TweetMate-for-Android/blob/0a5013e538213bacb2774f24bfa4ed6959fa720c/TweetMate/src/main/java/com/ybsystem/tweetmate/activities/VideoActivity.java#L111"></span>Switch statement on an <code>int</code> with known associated constant missing case <code>Player.STATE_BUFFERING</code>, <code>Player.STATE_IDLE</code>, <code>Player.STATE_READY</code></td>
    </tr>
  </tbody>
</table>